google-cloud-logging/lib/google/cloud/logging/logger.rb

```diff
@@ -534,6 +544,8 @@ def silence temp_level = ::Logger::ERROR
 
         ##
         # @private Write a log entry to the Stackdriver Logging service.
+        # The expectation is that this method is always called from within a
+        # synchronize block.
```

Comment:
Then it might be safer just to synchronize this method rather than depending on the caller.

And actually, I would rather not synchronize this entire method, because it could block for an entire API round trip if we're not using the AsyncWriter. Better would be to pull out the needed values out of `info` (and just synchronize that section) and then pass those values individually to `entry_labels`.

Reply:
I reverted it back to to using the `request_info` method, which uses `synchronize` to access the hash. Since that is the only time the hash is touched, that means we can remove the `synchronize` block from the call to `write_entry`.